### DISCUSSION TITLE
Authentication Policy Not Working

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#6066] [Questions] [answered]
I have a somewhat complex authentication policy that is trying to limit the projects that a user has access to. The auth.uid() matches the user_id on profiles which goes through a many-to-many table _profilesToteams which matches to the projects team_id. Here is the authentication policy:

```
(auth.uid() IN ( SELECT profiles.user_id
   FROM profiles,
    "_profilesToteams"
  WHERE ((profiles.id = "_profilesToteams"."A") AND ("_profilesToteams"."B" = projects.team_id))))
``` 
  
If I run the query and provide a team_id:
 
```
SELECT profiles.user_id
   FROM profiles,
    "_profilesToteams"
  WHERE ((profiles.id = "_profilesToteams"."A") AND ("_profilesToteams"."B" = 'ef1f2484-0aee-4030-bec3-8481586dcc84'))
```
  
 It returns one user_id, but when I run the app locally and make a query through the Supabase client, all projects are returned.
 
 I fear I may not be understanding how this works exactly and my policy should be revised. Any help would be appreciated.

**Comment by jaredcosulich** [comment]
Actually do the authentication policies work locally? I just set the policy to return false:

```
CREATE POLICY "Disable access to all users" ON public.projects FOR ALL USING (false);
```

But this query:

```
const { data, error } = await supabaseClient
    .from('projects')
    .select(`
      id,
      title,
      teams(id, title)
    `)
```

Still returns all projects. Am I misunderstanding something?

**Comment by GaryAustin1** [comment, marked as answer]
Hate to ask first obvious question, besides having a policy is RLS enabled for the table?
Are you using service key instead of anon key when setting up client?

**Reply by jaredcosulich** [reply]
Yes, just double checked and it is the anon key

**Reply by burggraf** [reply]
Also,  I think his first question was this:  "do you have RLS turned ON (enabled) for that table"?

You can quickly find this in the dashboard under Authentication / Policies.

**Reply by jaredcosulich** [reply]
Ugh, yes, this was the problem. We're adding RLS policies via sql migration when we launch locally but the RLS was not enabled. From the documentation it looks like we also need to run this on each table that has a policy:

```
alter table profiles
  enable row level security;
```

I've added that and it seems to be working now. Thank you for the help.

**Reply by burggraf** [reply]
Awesome!  Thanks for following up.
